internal/auth/auth.go

```diff
@@ -0,0 +1,47 @@
+package auth
```

Comment:
I think `auth` is a bit too generic. WDYT about `registryauth`?

Reply:
Or we could move it to the `registry` package. `registry.Auth` looks good to me.

Reply:
I liked the fact that `auth` is a bit generic so it can be reused and extended also in other contexts such as interacting with certs configMaps in the future.

For this reason I also would like to to make it a nested package of the `registry` pkg.

Reply:
> I liked the fact that `auth` is a bit generic

We're only talking about authentication against a registry, right? I don't think we would use push secrets for any other purpose.

> it can be reused and extended also in other contexts such as interacting with certs configMaps in the future

You mean CA certificates? Those are not about authenticating but securing the connection to the registry.

> For this reason I also would not like to to make it a nested package of the `registry` pkg.

I think it would be very appropriate for this code to be there.

Reply:
As agreed, we will keep it as an `auth` internal package, and the interface will be renamed to `RegistryAuth`. In the future, this package can be extended with additional interfaces such as `GithubAuth` etc.